### PR TITLE
update RSPECs for 8.6.1 release

### LINKED ISSUE #3285: Change issue S3011 type to code smell
OOP visibility/accessibility is likely more a code quality subject than security thus [S3011](https://jira.sonarsource.com/browse/RSPEC-3011) should live as a code smell

Related to:
https://github.com/SonarSource/sonar-dotnet/issues/1983

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
Update RSPEC files before 8.6.1 bugfix release.

### CODE REVIEW

File: sonaranalyzer-dotnet/rspec/cs/S3011_c#.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "tags": [
     "owasp-a3"
   ],
-  "defaultSeverity": "Critical",
+  "defaultSeverity": "Minor",
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/rspec/cs/S3011_c#.json:12]: Indeed RSPEC-3011 is not related to security and it's a code smell but do we really want to make it minor?

Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/rspec/cs/S3011_c#.json:12]: It has an impact on both code quality and performance as reflection is usually much slower.

File: sonaranalyzer-dotnet/rspec/cs/S4834_c#.html [hunk 2]
```diff
@@ -76,5 +75,9 @@ <h2>See</h2>
 <ul>
   <li> <a href="https://www.owasp.org/index.php/Top_10-2017_A5-Broken_Access_Control">OWASP Top 10 2017 Category A5</a> - Boken Access Control </li>
   <li> <a href="https://www.sans.org/top25-software-errors/#cat3">SANS Top 25</a> - Porous Defenses </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/276.html">MITRE, CWE-276</a> - Incorrect Default Permissions </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/732.html">MITRE, CWE-732</a> - Incorrect Permission Assignment for Critical Resource </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/668.html">MITRE, CWE-668</a> -Exposure of Resource to Wrong Sphere </li>
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/rspec/cs/S4834_c#.html:80]: pedantinc: 
- there is a missing space at " -Exposure of Resource"
- there are some trailing spaces in the newly added xml elements

File: sonaranalyzer-dotnet/rspec/vbnet/S3011_vb.net.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
   "tags": [
     "owasp-a3"
   ],
-  "defaultSeverity": "Critical",
+  "defaultSeverity": "Minor",
   "ruleSpecification": "RSPEC-3011",
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/rspec/vbnet/S3011_vb.net.json:13]: Same comment about severity for VB.Net

File: sonaranalyzer-dotnet/rspec/cs/S3011_c#.json (hunk 1 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/rspec/cs/S3011_c#.json:12]: indeed, I will revert this as we won't change it now

but [using reflection](https://jira.sonarsource.com/browse/RSPEC-3011) should not be minor

File: sonaranalyzer-dotnet/rspec/vbnet/S4834_vb.net.html [hunk 4]
```diff
@@ -72,5 +71,9 @@ <h2>See</h2>
 <ul>
   <li> <a href="https://www.owasp.org/index.php/Top_10-2017_A5-Broken_Access_Control">OWASP Top 10 2017 Category A5</a> - Boken Access Control </li>
   <li> <a href="https://www.sans.org/top25-software-errors/#cat3">SANS Top 25</a> - Porous Defenses </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/276.html">MITRE, CWE-276</a> - Incorrect Default Permissions </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/732.html">MITRE, CWE-732</a> - Incorrect Permission Assignment for Critical Resource </li>
+  <li> <a href="https://cwe.mitre.org/data/definitions/668.html">MITRE, CWE-668</a> -Exposure of Resource to Wrong Sphere </li>
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/rspec/vbnet/S4834_vb.net.html:76]: same comment as on C# code

**Comment by andrei-epure-sonarsource** [comment]
@costin-zaharia-sonarsource I reverted the changes to S3011 and S4834 because the decrease of severity needs to be double checked with AppSec

File: sonaranalyzer-dotnet/rspec/cs/S4564_c#.json [hunk 5]
```diff
@@ -7,20 +7,10 @@
     "constantCost": "5min"
   },
   "tags": [
-    "cwe",
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/rspec/cs/S4564_c#.json:10]: this rule https://jira.sonarsource.com/browse/RSPEC-4564 got deprecated, hence the removal of standards

**Pull request merged**